packages/truffle-compile/index.js

```diff
@@ -280,72 +280,39 @@ function replaceLinkReferences(bytecode, linkReferences, libraryName) {
   return bytecode;
 }
 
-function orderABI(contract) {
-  var contract_definition;
-  var ordered_function_names = [];
-
-  for (var i = 0; i < contract.legacyAST.children.length; i++) {
-    var definition = contract.legacyAST.children[i];
-
-    // AST can have multiple contract definitions, make sure we have the
-    // one that matches our contract
-    if (
-      definition.name !== "ContractDefinition" ||
-      definition.attributes.name !== contract.contract_name
-    ) {
-      continue;
-    }
-
-    contract_definition = definition;
-    break;
+function orderABI({ abi, contract_name: contractName, ast }) {
+  // AST can have multiple contract definitions, make sure we have the
+  // one that matches our contract
+  const contractDefinition = ast.nodes.filter(
```

Comment:
I'd suggest using `arr.find(...)` rather than `arr.filter(...)[0]`.

Reply:
That's a thing!? Heck yeah!